### DISCUSSION TITLE
Grab ANY/ALL custom properties of an element

### DISCUSSION THREAD

**Discussion by benjamin-lindsey** [iTwin/itwinjs-core#5981] [Q&A] [answered]
Hi,

Is there a straightforward way of grabbing any/all properties of an element? This would include custom item type properties.

Taking an example from the iTwin accreditation course, we had to know ahead of time the SmartDevice item type, part of the house schema, in order to query for smartDeviceId, smartDeviceType, etc.

So how can I extract properties of an element we know nothing about except for its ECInstanceID and its other properties returned by `SELECT *`?

Thanks for your help,
Ben

**Comment by benjamin-lindsey** [comment]
Hmm, perhaps what I'm looking for is the Presentation library?

**Comment by pmconne** [comment, marked as answer]
Have you looked at [PresentationManager.getElementProperties](https://www.itwinjs.org/reference/presentation-backend/core/presentationmanager/getelementproperties/)?
You can call it from [the frontend](https://www.itwinjs.org/reference/presentation-frontend/core/presentationmanager/#getelementproperties) too.
You may find some useful ideas in [this discussion](https://github.com/iTwin/itwinjs-core/discussions/3343).

**Comment by benjamin-lindsey** [comment]
@pmconne - Perfect, thanks Paul!
